src/Microsoft.Restier.Providers.EntityFramework/Submit/ChangeSetInitializer.cs

```diff
@@ -40,6 +40,13 @@ internal class ChangeSetInitializer : IChangeSetInitializer
             {
                 object strongTypedDbSet = dbContext.GetType().GetProperty(entry.EntitySetName).GetValue(dbContext);
                 Type entityType = strongTypedDbSet.GetType().GetGenericArguments()[0];
+
+                // This means request entity is sub type of entity type
+                if (entry.ActualEntityType != null && entityType != entry.ActualEntityType)
```

Comment:
In delete case, it will be null now. We do not set the value for delete case.
I add this also not breaking any existing logic like test cases which does not have type inheritance and does not set this.
Let me know if you still prefer some enforcement.